source/extensions/stat_sinks/hystrix/hystrix.h

```diff
@@ -0,0 +1,144 @@
+
+private:
+  /**
+   * Format the given key and absl::string_view value to "key"="value", and adding to the
+   * stringstream.
+   */
+  void addStringToStream(absl::string_view key, absl::string_view value, std::stringstream& info);
+
+  /**
+   * Format the given key and uint64_t value to "key"=<string of uint64_t>, and adding to the
+   * stringstream.
+   */
+  void addIntToStream(absl::string_view key, uint64_t value, std::stringstream& info);
+
+  /**
+   * Format the given key and value to "key"=value, and adding to the stringstream.
+   */
+  void addInfoToStream(absl::string_view key, absl::string_view value, std::stringstream& info);
+
+  /**
+   * Generate HystrixCommand event stream.
+   */
+  void addHystrixCommand(std::stringstream& ss, absl::string_view cluster_name,
+                         uint64_t max_concurrent_requests, uint64_t reporting_hosts,
+                         uint64_t rolling_window);
+
+  /**
+   * Generate HystrixThreadPool event stream.
+   */
+  void addHystrixThreadPool(std::stringstream& ss, absl::string_view cluster_name,
+                            uint64_t queue_size, uint64_t reporting_hosts, uint64_t rolling_window);
+
+  /**
+   * Building lookup name map for all specific cluster values.
+   */
+  void CreateCounterNameLookupForCluster(const std::string& cluster_name);
+
+  RollingStatsMap rolling_stats_map_;
+  uint64_t current_index_;
+  const uint64_t window_size_;
+  // TODO(trabetti): do we want this to be configurable through the HystrixSink in config file?
+  static const uint64_t DEFAULT_NUM_OF_BUCKETS = 10;
+  std::map<std::string, std::map<std::string, std::string>> counter_name_lookup;
+};
+
+typedef std::unique_ptr<HystrixStatCache> HystrixStatCachePtr;
+
+namespace Hystrix {
+
+class HystrixSink : public Stats::Sink, public Logger::Loggable<Logger::Id::hystrix> {
+public:
+  HystrixSink(Server::Instance& server, uint64_t num_of_buckets);
+  HystrixSink(Server::Instance& server);
+  Http::Code handlerHystrixEventStream(absl::string_view, Http::HeaderMap& response_headers,
+                                       Buffer::Instance&, Server::AdminFilter& admin_filter);
+  void init();
+  void beginFlush() override;
+  void flushCounter(const Stats::Counter&, uint64_t) override{};
+  void flushGauge(const Stats::Gauge&, uint64_t) override{};
+  void endFlush() override;
+  void onHistogramComplete(const Stats::Histogram&, uint64_t) override{};
+
+  /**
+   * register a new connection
+   */
+  void registerConnection(Http::StreamDecoderFilterCallbacks* callbacks_to_register);
+  /**
+   * remove registered connection
+   */
+  void unregisterConnection(Http::StreamDecoderFilterCallbacks* callbacks_to_remove);
+  HystrixStatCache& getStats() { return *stats_; }
+
+private:
+  HystrixStatCachePtr stats_;
+  std::vector<Http::StreamDecoderFilterCallbacks*> callbacks_list_{};
+  Server::Instance& server_;
+  std::map<std::string, uint64_t> current_stat_values_;
```

Comment:
Can you comment here why you don't have a struct for matching stat-name to value?  It wasn't obvious from first glance of the code.

On my first read of the code I thought it wasn't necessary, but then I found that you are printing the names & values, so it's easier to have them stored this way.

Also comment on why you need this map ordered.  Most maps in Envoy are unordered.  I can tell from looking at the code that this makes the ordering predictable, which is good.  I'm just asking for comments here.

Reply:
Is `current_stat_values_` being used anywhere?  I'm having trouble finding references to it...?

Reply:
yes, in source/extensions/stat_sinks/hystrix/hystrix.cc which github now skips in /files until you click through.

Reply:
Sorry, I should've been more clear.  All I see is that it's being cleared in the `beginFlush()` method, but I don't see it being used anywhere else.

Reply:
@jmarantz why does hysrtix.cc being skipped? I also noticed that some files are not updated here. Is it because the github PR capacity issue?